avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerializer.java

```diff
@@ -46,6 +46,12 @@
 
     try {
       schema = getSchema(object);
+      boolean enableAutoSchemaRegistry = schemaRegistry.isAutoSchemaRegistryEnabled();
+
+      if(!enableAutoSchemaRegistry && schemaRegistry.getVersion(subject,schema) <=0 ) {
```

Comment:
This behavior looks right, but the implementation needs to be adjusted since `schemaRegistry.getVersion(...)` throws a RestClientException if the schema is not in the registry (rather than returning a value <= 0)